### DISCUSSION TITLE
Query structs with Vectors — wait what?

### DISCUSSION THREAD

**Discussion by thomasmost** [tokio-rs/axum#1964] [Q&A] [answered]
Forgive me being stupid, I recognize I should be able to figure this out from this: https://github.com/tokio-rs/axum/issues/434#issuecomment-954898159

...but is this actually supported for any particular url encoding? Can someone provide an example of a valid query vector as represented in the request URI?

**Comment by davidpdrsn** [comment, marked as answer]
There is no standard way to encode query params as sequences, so axum's default `Query` extractor doesn't support it.

You'll wanna use https://docs.rs/axum-extra/latest/axum_extra/extract/struct.Query.html (notice this is in axum-extra) or https://crates.io/crates/serde_qs (which have axum support) instead.

**Reply by thomasmost** [reply]
Got it, thank you for the quick clarification 👍🏼
